From 2f0652585626fbc311422e89dcfbaabab440767d Mon Sep 17 00:00:00 2001
From: Luo Meng <luomeng12@huawei.com>
Date: Thu, 13 Oct 2022 15:57:12 +0800
Subject: [PATCH 0746/1076] [Huawei] Revert "[Backport] cifs: fix double free
 race when mount fails in cifs_get_root()"

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 187710

--------------------------------

This reverts commit 4370af0cfd776e3ef575b27a93a57dd3d02a6ecb.

Commit 2fe0e281f7ad witch merged by LTS (cifs: fix double free race
when mount fails in cifs_get_root()) fixes a double free. However,
the previous patch d17abdf75665 is not merged. There is no such
issue on 5.10 because it will return after cifs_cleanup_volume_info().

Since merge this patch, cifs_cleanup_volume_info() is skipped, leading
to a memory leak.

Signed-off-by: Luo Meng <luomeng12@huawei.com>
---
 fs/cifs/cifsfs.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/fs/cifs/cifsfs.c b/fs/cifs/cifsfs.c
index bc957e6ca48b..6e1e7e44706c 100644
--- a/fs/cifs/cifsfs.c
+++ b/fs/cifs/cifsfs.c
@@ -864,7 +864,6 @@ cifs_smb3_do_mount(struct file_system_type *fs_type,
 
 out_super:
 	deactivate_locked_super(sb);
-	return root;
 out:
 	cifs_cleanup_volume_info(volume_info);
 	return root;
-- 
2.17.1

